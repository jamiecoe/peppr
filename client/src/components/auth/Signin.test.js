import React from 'react';
import { shallow } from 'enzyme';
import { SignIn } from './Signin';

describe('SignIn', () => {
  const mockResetError = jest.fn();
  const mockSigninUser = jest.fn();
  const mockHandleSubmit = jest.fn(fn => fn);
  const props = {
    resetError: mockResetError,
    signinUser: mockSigninUser,
    handleSubmit: mockHandleSubmit,
    error: '',
  };
  let signin = shallow(<SignIn {...props} />, {
    disableLifecycleMethods: false,
  });

  it('should render properly', () => {
    expect(signin).toMatchSnapshot();
  });

  it('should call `resetError` in componentDidMount()', () => {
    expect(mockResetError).toHaveBeenCalled();
  });

  describe('when the user submits the form', () => {
    beforeEach(() => {
      signin.find('form').simulate('submit');
    });

    it('should call mockHandleSubmit()', () => {
      expect(mockHandleSubmit).toHaveBeenCalled();
    });

    it('should call mockSigninUser', () => {
      expect(mockSigninUser).toHaveBeenCalled();
    });
  });

  describe('when there is an error on props', () => {
    beforeEach(() => {
      props.error = 'this is a test error';
      signin = shallow(<SignIn {...props} />);
    });

    it('should render the error message from props', () => {
      expect(signin.find('.signin-form-error-msg').text()).toBe(
        `Oops! ${props.error}`,
      );
    });
  });

  describe('when the `renderField` method is invoked', () => {
    const fieldProps = {
      meta: {
        touched: true,
        error: 'test error',
      },
      textfield: true,
      arialabel: 'test arialabel',
      placeholder: 'test placeholder',
      input: {},
    };

    const RenderedField = signin.instance().renderField(fieldProps)[1];
    const renderField = shallow(RenderedField);

    it('should render properly', () => {
      expect(renderField).toMatchSnapshot();
    });

    it('should include an error message', () => {
      expect(
        renderField
          .find('.landing__input--errortext')
          .text(),
      ).toBe(fieldProps.meta.error);
    });
  });
});
