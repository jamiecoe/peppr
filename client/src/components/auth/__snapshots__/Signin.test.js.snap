// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    error=""
    handleSubmit={[Function]}
    resetError={[Function]}
    signinUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <Field
                    arialabel="Email"
                    component={[Function]}
                    name="email"
                    placeholder="Email"
                    type="email"
          />
          <Field
                    arialabel="Password"
                    component={[Function]}
                    name="password"
                    placeholder="Password"
                    type="password"
          />
          <p
                    className="landing__input--errortext"
          >
                    <span />
          </p>
          <input
                    className="landing__form__btn"
                    defaultValue="Login"
                    type="submit"
          />
</form>,
        <Link
          className="landing__form__togglelink"
          replace={false}
          to="/signup"
>
          New to Peppr? Sign up
</Link>,
      ],
      "className": "landing__form__container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Field
              arialabel="Email"
              component={[Function]}
              name="email"
              placeholder="Email"
              type="email"
/>,
            <Field
              arialabel="Password"
              component={[Function]}
              name="password"
              placeholder="Password"
              type="password"
/>,
            <p
              className="landing__input--errortext"
>
              <span />
</p>,
            <input
              className="landing__form__btn"
              defaultValue="Login"
              type="submit"
/>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "arialabel": "Email",
              "component": [Function],
              "name": "email",
              "placeholder": "Email",
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "arialabel": "Password",
              "component": [Function],
              "name": "password",
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span />,
              "className": "landing__input--errortext",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "landing__form__btn",
              "defaultValue": "Login",
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "New to Peppr? Sign up",
          "className": "landing__form__togglelink",
          "replace": false,
          "to": "/signup",
        },
        "ref": null,
        "rendered": "New to Peppr? Sign up",
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <Field
                        arialabel="Email"
                        component={[Function]}
                        name="email"
                        placeholder="Email"
                        type="email"
            />
            <Field
                        arialabel="Password"
                        component={[Function]}
                        name="password"
                        placeholder="Password"
                        type="password"
            />
            <p
                        className="landing__input--errortext"
            >
                        <span />
            </p>
            <input
                        className="landing__form__btn"
                        defaultValue="Login"
                        type="submit"
            />
</form>,
          <Link
            className="landing__form__togglelink"
            replace={false}
            to="/signup"
>
            New to Peppr? Sign up
</Link>,
        ],
        "className": "landing__form__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Field
                arialabel="Email"
                component={[Function]}
                name="email"
                placeholder="Email"
                type="email"
/>,
              <Field
                arialabel="Password"
                component={[Function]}
                name="password"
                placeholder="Password"
                type="password"
/>,
              <p
                className="landing__input--errortext"
>
                <span />
</p>,
              <input
                className="landing__form__btn"
                defaultValue="Login"
                type="submit"
/>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "arialabel": "Email",
                "component": [Function],
                "name": "email",
                "placeholder": "Email",
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "arialabel": "Password",
                "component": [Function],
                "name": "password",
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span />,
                "className": "landing__input--errortext",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "landing__form__btn",
                "defaultValue": "Login",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "New to Peppr? Sign up",
            "className": "landing__form__togglelink",
            "replace": false,
            "to": "/signup",
          },
          "ref": null,
          "rendered": "New to Peppr? Sign up",
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": false,
  },
}
`;

exports[`SignIn when the \`renderField\` method is invoked should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="landing__input--errortext"
>
    test error
</div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "2",
    "nodeType": "host",
    "props": Object {
      "children": "test error",
      "className": "landing__input--errortext",
    },
    "ref": null,
    "rendered": "test error",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "2",
      "nodeType": "host",
      "props": Object {
        "children": "test error",
        "className": "landing__input--errortext",
      },
      "ref": null,
      "rendered": "test error",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
